src/main/scala/play/api/cache/redis/connector/AkkaSerializer.scala

```diff
@@ -49,15 +49,13 @@ trait AkkaSerializer {
 private[ connector ] class AkkaEncoder( serializer: Serialization ) {
 
   /** Unsafe method encoding the given value into a string */
-  def encode( value: Any ): String = value match {
-    // null is special case
-    case null => unsupported( "Null is not supported by the redis cache connector." )
+  def encode( value: Any ): String = Option(value) match {
```

Comment:
You are missing the point here and introduce a very major overhead instead. The `null` value should be never present. If so, it's a bug.

Matching of the `value` and `null` is quite efficient as it is a simple test against a null value, which introduces a very low overhead and then we can move to the actual business.

However, you suggest:

- first, call `Option.apply`, which also **includes the test against null**
- second, construct a new instance of `Some`. `None` should be never used.
- third, you need to always check that it is instance of `Some` and then call `unapply` just to access the actual `value` inside the `Option`

This is much bigger overhead without any added value.